src/diagnostics/logPointsManager.ts

```diff
@@ -0,0 +1,298 @@
+        return new WebSiteManagementClient(this.azureAccount.getCredentialByTenantId(this.subscription.tenantId), this.subscription.subscriptionId);
+    }
+
+    public async fetchPublishCrentential(site: WebSiteModels.Site): Promise<WebSiteModels.User> {
+        const siteClient = new WebSiteManagementClient(this.azureAccount.getCredentialByTenantId(this.subscription.tenantId), this.subscription.subscriptionId);
+        const user = await util.getWebAppPublishCredential(siteClient, site);
+        this._cachedPublishCredential = user;
+        return user;
+    }
+
+    public async getCachedCredentialOrRefetch(site: WebSiteModels.Site): Promise<WebSiteModels.User> {
+        if (this.lastUsedPublishCredential) {
+            return Promise.resolve(this.lastUsedPublishCredential);
+        }
+
+        return this.fetchPublishCrentential(site);
+    }
+
+    protected beforeExecute() { }
+
+    protected onExecuteError(error: Error) {
+        if (error instanceof UserCancelledError) {
+            return;
+        }
+        this.writeline(`Starting Log Points Session failed - ${error.message}`);
+        this.writeline('');
+    }
+}
+
+class PromptSlotSelection extends WizardStep {
+    constructor(private _wizard: LogPointsSessionAttach, readonly site: WebSiteModels.Site) {
+        super(_wizard, 'Choose a deployment slot.');
+    }
+
+    public async execute(): Promise<void> {
+        let deploymentSlots: WebSiteModels.Site[];
+
+        // Decide if this AppService uses deployment slots
+        await vscode.window.withProgress({ location: vscode.ProgressLocation.Window }, async p => {
+            let message = 'Enumerating deployment slots for the App Service...';
+            p.report({ message: message });
+            this._wizard.writeline(message);
+            deploymentSlots = await this.getDeploymentSlots();
+        });
+
+        this._wizard.writeline(`Got ${deploymentSlots.length} deployment slot(s)`);
+
+        // if there is only one slot, just use that one and don't prompt for user selection.
+        if (deploymentSlots.length == 1) {
+            this._wizard.selectedDeploymentSlot = deploymentSlots[0];
+            this._wizard.writeline(`Automatically selected deployment solt ${this._wizard.selectedDeploymentSlot.name}.`);
+            return;
+        }
+
+
+        const deploymentQuickPickItems = deploymentSlots.map((deploymentSlot: WebSiteModels.Site) => {
+            return <QuickPickItemWithData<WebSiteModels.Site>>{
+                label: util.extractDeploymentSlotName(deploymentSlot) || deploymentSlot.name,
+                description: '',
+                data: deploymentSlot
+            }
+        });
+
+        const quickPickOption = { placeHolder: `Please select a deployment slot: (${this.stepProgressText})` };
+        const pickedItem = await this.showQuickPick(deploymentQuickPickItems, quickPickOption);
+
+        this._wizard.selectedDeploymentSlot = pickedItem.data;

```

Comment:
What if user cancels the prompt?

Reply:
Will capture such case and give user a prompt.